src/tasks/python/PythonTaskHelper.ts

```diff
@@ -0,0 +1,193 @@
+
+        // The debugger will complain if the logs directory does not exist.
+        if (!fse.existsSync(dbgLogsFolder)) {
+            fse.emptyDirSync(dbgLogsFolder);
+        }
+
+        // User input is honored in all of the below.
+        runOptions.volumes = this.inferVolumes(runOptions, launcherFolder, dbgLogsFolder);
+        runOptions.ports = this.inferPorts(runOptions, helperOptions);
+        runOptions.entrypoint = runOptions.entrypoint || "python";
+        runOptions.command = runOptions.command || launcherCommand;
+
+        runOptions.env = this.addDebuggerEnvironmentVars(runOptions.env);
+        runOptions.portsPublishAll = runOptions.portsPublishAll || true;
+
+        return runOptions;
+    }
+
+    public async provideDockerBuildTasks(context: DockerTaskScaffoldContext): Promise<DockerBuildTaskDefinition[]> {
+        return [
+            {
+                type: "docker-build",
+                label: "docker-build",
+                platform: "python",
+                dockerBuild: {
+                    tag: getDefaultImageName(context.folder.name),
+                    dockerfile: unresolveWorkspaceFolder(context.dockerfile, context.folder),
+                    /* eslint-disable no-template-curly-in-string */
+                    context: "${workspaceFolder}"
+                }
+            }
+        ];
+    }
+
+    public async provideDockerRunTasks(context: DockerTaskScaffoldContext, options: PythonScaffoldingOptions): Promise<DockerRunTaskDefinition[]> {
+        const runOptions: PythonTaskRunOptions = {
+            args: inferPythonArgs(options.projectType, context.ports)
+        };
+
+        if ((options.target as PythonFileTarget).file) {
+            runOptions.file = (options.target as PythonFileTarget).file;
+        } else {
+            runOptions.module = (options.target as PythonModuleTarget).module;
+        }
+
+        return [{
+            type: "docker-run",
+            label: "docker-run: debug",
+            dependsOn: ["docker-build"],
+            python: runOptions
+        }];
+    }
+
+    private inferVolumes(runOptions: DockerRunOptions, launcherFolder: string, dbgLogsFolder: string): DockerContainerVolume[] {
+        const volumes: DockerContainerVolume[] = [];
+
+        if (runOptions.volumes) {
+            for (const volume of runOptions.volumes) {
+                volumes.push(volume);
+            }
+        }
+
+        const dbgVolumes: DockerContainerVolume[] = [
+            {
+                localPath: launcherFolder,
+                containerPath: "/pydbg",
+                permissions: "ro"
+            },
+            {
+                localPath: dbgLogsFolder,
+                containerPath: "/dbglogs",
+                permissions: "rw"
+            }];
+
+        dbgVolumes.map(dbgVol => {
+            if (volumes.find(volume => volume.containerPath === dbgVol.containerPath) === undefined) {
+                volumes.push(dbgVol);
```

Comment:
Over in `NetCoreTaskHelper` I have an `addVolumeWithoutConflicts` which will add volumes to a list, while never overriding something the user specified. Should we refactor that method out of `NetCoreTaskHelper` and use it here?